test/controllers/gems_controller_test.rb

```diff
@@ -12,4 +12,11 @@ def setup
     assert_template :index
     assert_select "title", "My Gems | RubyGems AdoptionCenter"
   end
+
+  test "should list gems" do
+    gems = return_gems
+
+    assert_equal Array, gems.class
+    assert_equal 'gem1', gems.first['name']
+  end
```

Comment:
Check this test out as an example: https://github.com/rubygems/rubygems.org/blob/master/test/integration/gems_test.rb